### PR TITLE
fix bug in linestring intersection when linestring is degenerate point

### PR DESCRIPTION
**Pull request by cookiedan42**
- [x] I agree to follow the project's [code of conduct](https://github.com/georust/geo/blob/main/CODE_OF_CONDUCT.md).
- [ ] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
---

Embarrassing that this edge case slipped through my earlier work 😞

### CODE REVIEW

File: geo/src/algorithm/intersects/line_string.rs [hunk 1]
```diff
@@ -86,6 +91,21 @@ mod test {
...
+        let pt: Point<f64> = wkt! {POINT(1 1)}.convert();
+        let poly: Polygon<f64> = wkt! {RECT(0 0, 1 1)}.to_polygon().convert();
+        let ls1: LineString<f64> = wkt! {LINESTRING(1 1)}.convert();
+        let ls2: LineString<f64> = wkt! {LINESTRING(1 1, 1 1)}.convert();
+
+        assert_eq!(ls1.intersects(&pt), ls1.relate(&pt).is_intersects());
+        assert_eq!(ls2.intersects(&pt), ls2.relate(&pt).is_intersects());
+        assert!(ls1.intersects(&poly));
+        // TODO: uncomment when relate is fixed, this currently panics
+        // assert_eq!(ls1.intersects(&poly), ls1.relate(&poly).is_intersects());
```
Comment (michaelkirk) [inline geo/src/algorithm/intersects/line_string.rs:105]: Looks like you opened #1482 to track this. Thank you!

Good find. This seems like a bug in CoordinatePosition. I'm taking a look now.

**Review by michaelkirk — APPROVED** [review]
Much ink and blood has been spilled over degenerate geometries. Thank you for adding more testing.
